features/steps/jobs_steps.go

```diff
@@ -214,3 +246,103 @@ func (f *JobsFeature) iCallGETJobsidUsingTheGeneratedId() error {
+	f.callPostJobs()
+	f.callPostJobs()
+
+	return f.ErrorFeature.StepError()
+}
+
+//iWouldExpectThereToBeThreeOrMoreJobsReturnedInAList is a feature step that can be defined for a specific JobsFeature.
+//It checks the response from calling GET /jobs to make sure that a list containing three or more jobs has been returned.
+func (f *JobsFeature) iWouldExpectThereToBeThreeOrMoreJobsReturnedInAList() error {
+	f.responseBody, _ = ioutil.ReadAll(f.ApiFeature.HttpResponse.Body)
+
+	var response models.Jobs
+	err := json.Unmarshal(f.responseBody, &response)
+	if err != nil {
+		return err
+	}
+	numJobsFound := len(response.Job_List)
+	assert.True(&f.ErrorFeature, numJobsFound >= 3, "The list should contain three or more jobs but it only contains "+strconv.Itoa(numJobsFound))
+
+	return f.ErrorFeature.StepError()
+}
+
+//inEachJobIWouldExpectIdLast_updatedAndLinksToHaveThisStructure is a feature step that can be defined for a specific JobsFeature.
+//It checks the response from calling GET /jobs to make sure that each job contains the expected types of values of id,
+//last_updated, and links.
+func (f *JobsFeature) inEachJobIWouldExpectIdLast_updatedAndLinksToHaveThisStructure(table *godog.Table) error {
+	assist := assistdog.NewDefault()
+	expectedResult, err := assist.ParseMap(table)
+	if err != nil {
+		panic(err)
+	}
+	var response models.Jobs
+
+	err = json.Unmarshal(f.responseBody, &response)
+	if err != nil {
+		return err
+	}
+
+	for j := range response.Job_List {
+		job := response.Job_List[j]
+		err2 := f.checkStructure(err, job.ID, job.LastUpdated, expectedResult, job.Links)
+		if err2 != nil {
+			return err2
+		}
+
+	}
+
+	return f.ErrorFeature.StepError()
+}
+
+//eachJobShouldAlsoContainTheFollowingValues is a feature step that can be defined for a specific JobsFeature.
+//It checks the response from calling GET /jobs to make sure that each job contains the expected values of
+//all the remaining attributes of a job.
+func (f *JobsFeature) eachJobShouldAlsoContainTheFollowingValues(table *godog.Table) error {
+	expectedResult, err := assistdog.NewDefault().ParseMap(table)
+	if err != nil {
+		panic(err)
+	}
+	var response models.Jobs
+
+	_ = json.Unmarshal(f.responseBody, &response)
+
+	for j := range response.Job_List {
+		job := response.Job_List[j]
+		f.checkValuesInJob(expectedResult, job)
+	}
+
+	return f.ErrorFeature.StepError()
+}
+
+//theJobsShouldBeOrderedByLast_updatedWithTheOldestFirst is a feature step that can be defined for a specific JobsFeature.
+//It checks the response from calling GET /jobs to make sure that the jobs are in ascending order of their last_updated
+//times i.e. the most recently updated is last in the list.
+func (f *JobsFeature) theJobsShouldBeOrderedByLast_updatedWithTheOldestFirst() error {
+	var response models.Jobs
+	err := json.Unmarshal(f.responseBody, &response)
+	if err != nil {
+		return err
+	}
+	job_list := response.Job_List
+	timeToCheck := job_list[0].LastUpdated
+
+	j := 0
```

Comment:
lines 337, 338 and 345 can be reduced into 1 line to define for loop, see below (also made tweaks to other 1 liners in the for loop.

```
...
	for j := 1; j < len(job_list); j++ {
		index := strconv.Itoa(j-1)
		nextIndex := strconv.Itoa(j)
		nextTime := job_list[j].LastUpdated

		assert.True(&f.ErrorFeature, timeToCheck.Before(nextTime),
			"The value of last_updated at job_list["+index+"] should be earlier than that at job_list["+nextIndex+"]")

		timeToCheck = nextTime
	}
...
```

Reply:
yes that looks better - I should have thought about it more because it was a little messy before - done